atc/db/wall_test.go

```diff
@@ -0,0 +1,109 @@
+package db_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"time"
+)
+
+var _ = Describe("Wall", func() {
+	var (
+		testMessage string
+	)
+
+	BeforeEach(func() {
+		testMessage = "this is the test message!?"
+	})
+
+	Context(" a message is set", func() {
+		BeforeEach(func() {
+			err := wall.SetMessage(testMessage)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("with no expiration", func() {
+			message, err := wall.GetMessage()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(message).To(Equal(testMessage))
+		})
+		Context("with an expiration", func() {
+			It("the message has not expired", func() {
+				err := wall.SetExpiration(time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+
+				message, err := wall.GetMessage()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(message).To(Equal(testMessage))
+			})
+
+			It("the message has expired", func() {
+				err := wall.SetExpiration(time.Millisecond)
+				Expect(err).ToNot(HaveOccurred())
+
+				time.Sleep(time.Millisecond)
+
+				message, err := wall.GetMessage()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(message).To(Equal(""))
+			})
+		})
+	})
+
+	Context("multiple messages are set", func() {
+		It("returns the last message that was set", func() {
+			wall.SetMessage("test 1")
+			wall.SetMessage("test 2")
+			wall.SetMessage("test 3")
+
+			msg, err := wall.GetMessage()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(msg).To(Equal("test 3"))
+		})
+	})
+
+	Context("clear the message", func() {
+		It("returns no message", func() {
+			wall.SetMessage(testMessage)
+			msg, err := wall.GetMessage()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(msg).To(Equal(testMessage), "ensure the message has been set")
+
+			wall.Clear()
```

Comment:
```suggestion
			err = wall.Clear()
			Expect(err).ToNot(HaveOccurred())
```
just for consistency?